### PR TITLE
adding function for checking whether all nodes have launched

### PR DESCRIPTION
**Pull request by YoheiKakiuchi**
Adding functions for checking whether all nodes have launched.
One is function which can be called through python.
One is method for xmlrpc.

### CODE REVIEW

File: tools/roslaunch/src/roslaunch/netapi.py [hunk 1]
```diff
@@ -93,3 +93,29 @@ def list_processes(roslaunch_uris=None):
...
+        try:
+            r = xmlrpclib.ServerProxy(uri)
+            code, msg, val = r.get_registrations_complete()
+            if code == 1:
+                comps.append(val)
+        except:
+            #import traceback
+            #traceback.print_exc()
+            # don't have a mechanism for reporting these errors upwards yet
+            pass
```
Comment (dirk-thomas) [inline tools/roslaunch/src/roslaunch/netapi.py:120]: This looks a bit weird. By just passing the calling code can not even figure out which uri has actually been completed and which not.

**Comment by dirk-thomas** [comment]
Please also provide some more context for the patch. How should the new API be used in the future? Will you extend the patch to expose the functionality somehow?

**Comment by YoheiKakiuchi** [comment]
One big problem in roslaunch is that user can not decide the order to execute process.
We would like to use programs which were not written with ROS manner,
We would like to control timing to execute process.
So, it is useful to know all nodes launched in a launch file.

**Comment by dirk-thomas** [comment]
The "problem" in roslaunch about the unspecified startup order is an intentional design decision. Please see http://ros.org/wiki/roslaunch/Architecture#line-13 for more information.

**Comment by dirk-thomas** [comment]
Due to lack of response, the fact that the proposed API should not be exposed anywhere in the CLI and the inconsistent/incomplete API/return value (as commented above) this issue is getting closed.

If you have any further feedback please feel free to comment and/or update the pull request.

**Pull request closed without merging**